src/luascript.cpp

```diff
@@ -11683,13 +11683,13 @@ int LuaScriptInterface::luaConditionAddDamage(lua_State* L)
 int LuaScriptInterface::luaMonsterTypeCreate(lua_State* L)
 {
 	// MonsterType(id or name)
-	MonsterType* monsterType;
-	if (isNumber(L, 2)) {
-		monsterType = g_monsters.getMonsterType(getNumber<uint32_t>(L, 2));
-	} else {
-		monsterType = g_monsters.getMonsterType(getString(L, 2));
+	
+	if (!isString(L, 2)) {
+		lua_pushnil(L);
+		return 1;
 	}
-
+	auto monsterType = g_monsters.getMonsterType(getString(L, 2));
```

Comment:
I prefer to know the type instead of auto when it isn't too long, please don't overuse auto.